adagrams/game.py

```diff
@@ -1,11 +1,97 @@
+        else: 
+            dict_of_10_letters[letter] = 1
+            alphabet_dict[letter] -= 1
+            total_letters_dict += 1
+    return dict_of_10_letters
+
 def draw_letters():
-    pass
+    selected_dict = create_dict_of_10_letters()
+    list_of_letters = []
+    for letter, num in selected_dict.items():
+        for i in range(num):
+            list_of_letters.append(letter)
+    return list_of_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    copy_letter_bank = letter_bank.copy()
+    letters_from_word_list = list(word.upper())
+    while len(letters_from_word_list) > 0:
+        letter = letters_from_word_list[0]
+        if letter in copy_letter_bank:
+            copy_letter_bank.remove(letter)
+            letters_from_word_list.remove(letter)
+        else:
+            return False
+    return True
 
 def score_word(word):
-    pass
+    score_chart = {
+                    'A': 1, 'E': 1, 'I': 1, 'O': 1, 'L': 1, 'N': 1, 
+                    'R': 1, 'S': 1, 'T': 1, 'D': 2, 'G': 2,'B': 3, 
+                    'C': 3, 'M': 3, 'P': 3, 'F': 4, 'H': 4, 'V': 4, 
+                    'W': 4, 'Y': 4,'K': 5, 'J': 8, 'X': 8,'Q': 10, 
+                    'Z': 10
+                }
+    list_from_word = list(word.upper())
+    total_points = 0
+    for letter in list_from_word:
+        if letter in score_chart:
+            total_points += score_chart[letter]
+    if len(word) == 7 or len(word) == 8 or len(word) == 9 or len(word) == 10:
+        total_points += 8
+    return total_points
+
+#Esta f(x) va a tomar 2 tuples y devuelve la de mayor puntaje.
+#Con el valor que devuelve get_highest_word_score la va a comparar contra el ELEMENTO que tiene en el for en ese momento y retorna
+def find_max_of_2_tuples(tuple1, tuple2):
+    if tuple1 == None:
+        return tuple2
+    
+    score_tuple1 = tuple1[1]
+    score_tuple2 = tuple2[1]
+    len_tuple1 = len(tuple1[0])
+    len_tuple2 = len(tuple2[0])
+    if score_tuple1 == score_tuple2 and len_tuple1 == 10:
+        return tuple1
+    elif score_tuple1 == score_tuple2 and len_tuple2 == 10:
+        return tuple2
+    elif score_tuple1 == score_tuple2 and len_tuple1 < len_tuple2:
+        return tuple1
+    elif score_tuple1 == score_tuple2 and len_tuple1 == len_tuple2:
+        return tuple1
+    elif score_tuple1 > score_tuple2:
+        return tuple1
+    else: 
+        return tuple2
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    compare_scores = {}
+    for word in word_list:
+        score_per_word = score_word(word)
+        compare_scores[word] = score_per_word
+    
+    sorted_tuple_scores = sorted(compare_scores.items(), key=lambda x:x[1])
+    highest_score = None
+    for i in range(len(sorted_tuple_scores)):
+        tuple_element = sorted_tuple_scores[i]
+        check_higher_score = find_max_of_2_tuples(highest_score, tuple_element)
+        highest_score = check_higher_score
```

Comment:
Based on how you correctly compare tuples in `find_max_of_2_tuples`, it shouldn't actually matter if your list of scores is sorted or not!

So, looking at your code, I was curious about how necessary it is to have `sorted_tuple_scores = sorted(compare_scores.items(), key=lambda x:x[1])`.

The contents of `sorted_tuple_scores` and`compare_score` are actually identical! They contain the same information. The difference is that `sorted_tuple_scores` is an array of tuples, where `compare_score` is a dictionary.

If we change how we iterate, and iterate over the key-value pairs in `compare_score`, we can actually come to the same answer, without sorting or an extra list involved!

Consider this possible refactoring:

```python
    # sorted_tuple_scores = sorted(compare_scores.items(), key=lambda x:x[1])

    highest_score = None
    for word, score in compare_scores.items():
        tuple_element = (word, score)
        check_higher_score = find_max_of_2_tuples(highest_score, tuple_element)
        highest_score = check_higher_score
```